### PR TITLE
Add `async_nats::jetstream::Context::publish_with_headers`

### PR DESCRIPTION
**Pull request by caspervonb**

### CODE REVIEW

File: async-nats/tests/jetstream_tests.rs [hunk 1]
```diff
@@ -25,8 +25,37 @@ pub struct AccountInfo {
...
+        let stream = context
+            .create_stream(StreamConfig {
+                name: "TEST".to_string(),
+                subjects: vec!["foo".into(), "bar".into(), "baz".into()],
+                ..Default::default()
+            })
+            .await
+            .unwrap();
 
+        let headers = HeaderMap::new();
```
Comment (Jarema) [inline async-nats/tests/jetstream_tests.rs:47]: Note to us: when we add consumers / get raw message - add tests here.

**Comment by Jarema** [comment]
Maybe we want to have `publish_with_headers_or_options` that is called by both publish and publish with headers?
Options will surely come in soon and then there would be quite some duplication.

**Comment by Jarema** [comment]
@caspervonb looks good, but fails on fmt.

**Comment by caspervonb** [comment]
> @caspervonb looks good, but fails on fmt.
GitHub UI merge 😓

File: async-nats/src/jetstream/context.rs [hunk 2]
```diff
@@ -36,6 +38,31 @@ impl Context {
         }
     }
 
+    /// Publish a message
```
Comment (caspervonb) [inline async-nats/src/jetstream/context.rs:41]:
```suggestion
    /// Publish a message with headers
```

File: async-nats/tests/jetstream_tests.rs [hunk 3]
```diff
@@ -93,8 +122,7 @@ mod jetstream {
             "events".to_string()
         );
     }
-
-    #[tokio::test]
+
```
Comment (Jarema) [inline async-nats/tests/jetstream_tests.rs:125]: some weird formatting error? 🙃

**Review by Jarema — APPROVED** [review]
LGTM

**Pull request merged**